SOURCES/etc/xapi.d/plugins/smartctl.py

```diff
@@ -24,7 +24,7 @@ def get_information(session, args):
     with OperationLocker():
         disks = _list_disks()
         for disk in disks:
-            cmd = run_command(["smartctl", "-j", "-a", disk])
+            cmd = run_command(["smartctl", "-j", "-a", disk], check=False)
             results[disk] = json.loads(cmd['stdout'])
```

Comment:
Does smartctl always output JSON, even when there's an error? And what about stderr in case the command failed?

Reply:
I still would have checked the return code, and whenever the return code differs from 0 and 4, replace result[disk] with something which contains raw stdout and stderr. That's why I'd also check with XO so that we know how they would handle this case. Do they display the JSON, uninterpreted? Do they look for specific pieces of information?

Reply:
but we are already returning stdout that contains the error.

Reply:
they just display the raw json (and we see the error code in it)

Reply:
Ok, this, the man page and a test I made gives me all I wanted to know: it wraps the call to "regular" smartctl and apparently always puts the result inside JSON, even when there are non-recoverable errors (unless there's an error generating the JSON itself, I guess).

See this example where I have it a non recognised argument:
```
smartctl -j -a --wrongarg /dev/sdb
{
  "json_format_version": [
    1,
    0
  ],
  "smartctl": {
    "version": [
      7,
      0
    ],
    "svn_revision": "4883",
    "platform_info": "x86_64-linux-4.19.0+1",
    "build_info": "(local build)",
    "argv": [
      "smartctl",
      "-j",
      "-a",
      "--wrongarg",
      "/dev/sdb"
    ],
    "messages": [
      {
        "string": "=======> UNRECOGNIZED OPTION: wrongarg",
        "severity": "error"
      }
    ],
    "exit_status": 1
  }
}
```

So LGTM.